src/engine/LocalVocab.h

```diff
@@ -44,7 +64,11 @@ class LocalVocab {
 
  public:
   // Create a new, empty local vocabulary.
-  LocalVocab() = default;
+  LocalVocab(ad_utility::detail::AllocationMemoryLeftThreadsafe memoryLimit =
+                 ad_utility::makeAllocationMemoryLeftThreadsafeObject(
+                     ad_utility::MemorySize::megabytes(100)))
```

Comment:
For now, I would make the default `unlimited` as long as you don't pass it on (which will be a separate PR).